tests/verify_binary_contains_lints.rs

```diff
@@ -5,11 +5,7 @@ fn verify_binary_contains_lints() {
     let assert_on_crate_pair = |crate_pair: &str| {
         let mut cmd = Command::cargo_bin("cargo-semver-checks").unwrap();
         cmd.current_dir(format!("test_crates/{crate_pair}/new"))
-            .args([
-                "semver-checks",
-                "check-release",
-                "--baseline-root=../old/Cargo.toml",
-            ])
+            .args(["semver-checks", "check-release", "--baseline-root=../old/"])
```

Comment:
I've made this change temporarily so that the test passes locally, we should revert this change.

From what I understand, the flag `--baseline-root` was meant to take a directory to the project, but the old code somehow worked both when it received a directory and when it received a path to a manifest (and I believe that we didn't even know what we had this feature). Passing a directory is "more correct" in the sense that it aligns with the description from `--help`, but I think there might be some users who might rely on the accidental feature of passing manifests (like this test), so I think that we should make the current code also work for both directories and manifests.

Reply:
I also think it's reasonable to be able to extrapolate the directory if we're passed the manifest. Seems like an easy way to do the right thing even if the user did the wrong thing.

Nice catch! We got lucky this time, so let's add a test so we don't have to rely on luck for next time?

Reply:
OK, I've implemented it and I've added some tests. Also, the previous error messages of passing directory to `--manifest-path` were not clear. I'm not sure whether I made the right choice, but I've made it so it's possible to pass both a manifest and a directory path to the `--manifest-path` flag.